crates/hotshot/testing/src/consistency_task.rs

```diff
@@ -399,6 +406,23 @@ impl<TYPES: NodeType<BlockHeader = TestBlockHeader>, V: Versions> TestTaskState
                 timeout_task.abort();
             }
 
+            match qc2 {
+                Some(qc2) => {
+                    let last_leaf = &leaf_chain[0].leaf;
+                    ensure!(qc.view_number == last_leaf.view_number());
+                    ensure!(qc.data.leaf_commit == last_leaf.commit());
+                    ensure!(qc2.view_number == qc.view_number + 1);
+                },
+                None => {
+                    // Only the genesis decide is special in that it doesn't have a 2-chain of QCs
+                    // extending the last decided leaf.
+                    ensure!(
+                        view_number == TYPES::View::genesis(),
+                        "expected 2nd deciding QC for non-genesis decide"
+                    );
+                },
+            }
```

Comment:
Is there a better place in the test framework to hook in this new consistency check?